pytest_inmanta/plugin.py

```diff
@@ -1358,6 +1376,76 @@ def finalize_all_handlers(self) -> None:
         for handler_instance in self._handlers:
             self.finalize_handler(handler_instance)
 
+    def deploy_resource_v2(
+        self,
+        resource_type: str,
+        run_as_root: bool = False,
+        dry_run: bool = False,
+        status: Optional[const.ResourceState] = const.ResourceState.deployed,
+        **filter_args: object,
+    ) -> "DeployResultV2":
+        """
+        Deploy a resource of the given type, that matches the filter and assert the outcome
+
+        :param resource_type: the type of resource to deploy
+        :param filter_args: a set of kwargs, the resource must have all matching attributes set to the given values
+        :param run_as_root: run the handler as root or not
+        :param dry_run: only perform dryrun
+        :param status: expected status after deploy, set to None to not check
+        """
+        resource = self.get_resource(resource_type, **filter_args)
+        assert resource is not None, "No resource found of given type and filter args"
+
+        h = self.get_handler(resource, run_as_root)
+        assert h is not None
+
+        ctx = handler.HandlerContext(resource, dry_run=dry_run)
+        h.execute(ctx, resource, dry_run)
+        self.finalize_context(ctx)
+        self.finalize_handler(h)
+
+        out = DeployResultV2(resource=resource, ctx=ctx, handler=h)
+        if status is not None:
+            out.assert_status(status)
+
+        return out
+
+
+@dataclass
+class DeployResultV2:
+    ctx: HandlerContext
+    handler: ResourceHandler
+    resource: Resource
+
+    def assert_status(
+        self,
+        status: const.ResourceState = const.ResourceState.deployed,
+        change: const.Change = None,
+    ):
+        ctx = self.ctx
+        if ctx.status != status:
+            print("Deploy did not result in correct status")
+            print("Requested changes: ", ctx._changes)
```

Comment:
Why print? This will be lost if the test doesn't capture stdout, wich we usually not do

Reply:
I copy-pasted this. But I agree that logging would probably be better

Reply:
otoh print is always shown when the tests fail, logging requires explicitly setting the log level.

Reply:
opinion on this?

Reply:
I think it should be logs anyway